cirq/linalg/diagonalize.py

```diff
@@ -165,14 +179,15 @@ def bidiagonalize_real_matrix_pair_with_symmetric_products(
         ValueError: Matrices don't meet preconditions (e.g. not real).
     """
 
-    if np.any(np.imag(mat1) != 0):
-        raise ValueError('mat1 must be real.')
-    if np.any(np.imag(mat2) != 0):
-        raise ValueError('mat2 must be real.')
-    if not predicates.is_hermitian(mat1.dot(mat2.T), rtol=rtol, atol=atol):
-        raise ValueError('mat1 @ mat2.T must be symmetric.')
-    if not predicates.is_hermitian(mat1.T.dot(mat2), rtol=rtol, atol=atol):
-        raise ValueError('mat1.T @ mat2 must be symmetric.')
+    if check_preconditions:
+        if np.any(np.imag(mat1) != 0):
```

Comment:
is this safe from floating errors?

Reply:
The caller is supposed to send in a real matrix, not an approximately real matrix. In the actual context where this is used, that's done by sending in e.g. `np.real(u)` and `np.imag(u)` as the arguments.